lib/propolis/src/vmm/mem.rs

```diff
@@ -889,28 +888,44 @@ impl<'a, T: AsRef<[SubMapping<'a>]>> MappingExt for T {
             ));
         }
 
-        let mut total_capacity = 0;
+        let mut total_capacity: usize = 0;
         for mapping in self.as_ref().iter() {
-            total_capacity += mapping.len;
+            total_capacity =
+                total_capacity.checked_add(mapping.len).ok_or_else(|| {
+                    Error::new(
+                        ErrorKind::InvalidInput,
+                        "Total mapping too large",
```

Comment:
similarly, let's maybe say something like 
```suggestion
                        "Total length of all mappings would overflow a `usize`",